arches_component_lab/views/api/widgets.py

```diff
@@ -1,51 +1,48 @@
 import json
 
+from django.forms.models import model_to_dict
 from django.utils import translation
 from django.views.generic import View
 
 from arches.app.models import models
 from arches.app.utils.response import JSONResponse
 
 
-class WidgetConfigurationView(View):
-    def get(self, request, graph_slug, node_alias):
-        user_language = translation.get_language()
+def update_i18n_properties(response):
+    user_language = translation.get_language()
+    config = json.loads(response["config"].value)
 
-        node = models.Node.objects.get(
-            graph__slug=graph_slug, alias=node_alias, source_identifier_id__isnull=True
-        )
+    if "i18n_properties" in config and isinstance(config["i18n_properties"], list):
+        for prop in config["i18n_properties"]:
+            if (
+                prop in config
+                and isinstance(config[prop], dict)
+                and user_language in config[prop]
+            ):
+                config[prop] = config[prop][user_language]
+    response["config"] = config
+    return response
 
+
+class WidgetConfigurationView(View):
```

Comment:
I'd change this to something like `WidgetView` or `WidgetDataView`. Same with the node view